### ISSUE TITLE
Introspect/schema in production

### ISSUE THREAD

**Issue by arunoda** [apollostack/relay-standalone#2]
I hope Relay needs to know about the server schema to do some stuff, so I assume it needs to know about the schema.

I think in production, we cloud preload it with the initial HTML. Anyway, that's not a responsibility of this project.

For non-web apps, I think getting it via the introspect is the best option.

**Comment by stubailo** [comment]
Yeah, I think it should be possible to pass an existing schema object to the constructor, as listed here:

https://github.com/apollostack/relay-standalone#design

That way at least you don't need the initial roundtrip to get the schema.

I think in a full production setup where you know what server you are connecting to and you know the schema won't change, you can also pre-compile the queries, but I think that's a later optimization.

**Comment by smolinari** [comment]
> Should this be asynchronous because we want to introspect the server immediately? 

To be honest, I don't understand this question. 

> Do we even want to introspect the server in production?

Introspection is a tool for developers IMHO. Well, I can't currently imagine a use case where the normal user would need introspection. To me, introspection is used to answer the question, "What can I get from the server right now?", which only developers need to know. Right? So, it seems to me, using introspection should be a switch-on-able feature on a need to have basis. 

Scott
